### DISCUSSION TITLE
Question: Difference between new "ECS managed_draining" and the "settings.ecs.enable-spot-instance-draining"

### DISCUSSION THREAD

**Discussion by remiflament** [bottlerocket-os/bottlerocket#3746] [Q&A] [answered]
Hello,

ECS recently added [managed_draining](https://aws.amazon.com/about-aws/whats-new/2024/01/amazon-ecs-managed-instance-draining/) feature. It's **activated by default** on all new EC2 capacity providers.

AWS says, [in the documentation](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/managed-instance-draining.html#:~:text=Spot%20capacity%20rebalancing,state%20in%20response.), it can manage spot interruption by putting the instance in DRAINING state.

I was questioning if the settings [enable-spot-instance-draining](https://aws.amazon.com/about-aws/whats-new/2024/01/amazon-ecs-managed-instance-draining/) is now a duplicate, or should we still use it with managed_draining in an ec2 spot usage scenario?

Thank you,

**Comment by stockholmux** [comment]
@remiflament I think your second link should go to: https://bottlerocket.dev/en/os/1.19.x/api/settings/ecs/#enable-spot-instance-draining

**Comment by stockholmux** [comment, marked as answer]
So, the difference is that spot instance draining _only_ works for spot interruptions while managed draining works with several other types of interruptions. 

From the docs:

> Amazon ECS managed instance draining facilitates graceful shutdown of Amazon EC2 instances undergoing termination due to many reasons, not just Spot interruption

([Source](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/managed-instance-draining.html))


So it would only be useful if `managedDraining` is not `ENABLED` and you're using spot.

**Reply by remiflament** [reply]
Hello, yes thank you I ended with the same conclusion. I will just activate the AWS feature for my spot pool 👍🏻😁
